File: src/Peachpie.Avalonia/Layout/UxLayoutable.cs
using Avalonia.Layout;

namespace Peachpie.Avalonia.Layout;

public class UxLayoutable<T> : UxVisual<T> where T: Layoutable
{
    /// <summary>
    /// Gets or sets the element's preferred horizontal alignment in its parent.
    /// </summary>
    public HorizontalAlignment HorizontalAlignment
    {
        get => (HorizontalAlignment) WrappedObject.HorizontalAlignment;
        set => WrappedObject.HorizontalAlignment = (global::Avalonia.Layout.HorizontalAlignment) value;
    }

    /// <summary>
    /// Gets or sets the element's preferred vertical alignment in its parent.
    /// </summary>
    public VerticalAlignment VerticalAlignment
    {
        get => (VerticalAlignment) WrappedObject.HorizontalAlignment;
        set => WrappedObject.VerticalAlignment = (global::Avalonia.Layout.VerticalAlignment) value;
    }
}


public class UxLayoutable : UxLayoutable<Layoutable>
{
    
}

File: src/Samples/SharpPie/Peachpie/Avalonia/Layout/UxLayoutable.php
<?php

namespace Peachpie\Avalonia\Layout;

use Peachpie\Avalonia\UxVisual;

class UxLayoutable extends UxVisual
{
    /**
     * Defines how a control aligns itself horizontally in its parent control.
     *
     * @var HorizontalAlignment
     */
    public HorizontalAlignment $HorizontalAlignment;

    /**
     * Defines how a control aligns itself vertically in its parent control.
     *
     * @var VerticalAlignment
     */
    public VerticalAlignment $VerticalAlignment;
}